File: Problem 003/main.cpp
#include<iostream>

int main()
{
	int num, factor = 2;
	std::cout << "Enter a number: ";
	std::cin >> num;

	for (int temp = num; factor <= num / 2 && factor != temp; factor++)
	{
		while (temp % factor == 0 && factor != temp)
			temp /= factor;
	}
 
	std::cout << "Largest prime factor of " << num << " is " << factor;
}

File: Problem 003/main.py
import math

num = int(input("Enter a number: "))
temp = num

for factor in range(2, math.ceil(num / 2)):
    while (temp % factor == 0 and factor != temp):
        temp /= factor
    if (factor == temp):
        break

print("Largest prime factor of " + str(num) + " is " + str(factor))